### PR TITLE
[COST-3891] Worker for SWATCH Data Extraction

### PR DESCRIPTION
**Pull request by djnakabaale**
## Jira Ticket

[COST-3891](https://issues.redhat.com/browse/COST-3891)

## Description

This change will add the initial set up for SWATCH data extraction task worker.

## Testing

1. Checkout Branch
2. Restart Koku
3. Load test data
    ```
    make create-test-customer
    make load-test-customer-data
    ```
4. Hit the [celery_queue_lengths](http://127.0.0.1:5042/api/cost-management/v1/celery_queue_lengths/) endpoint.
    - You should see the subs extraction in the queues returned.
    ```
    {
        "download": 0,
        
        ...

        "subs_extraction": 0
    }


## Notes

...

### CODE REVIEW

**Review by maskarb — COMMENTED** [review]
Some initial comments

File: koku/subs/tasks.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    # Add additional accepted providers here
+)
+
+
+def check_schema_name(schema_name: str) -> str:
+    """Helper to check for and modify the schema name if needed."""
+    if schema_name and not schema_name.startswith(("acct", "org")):
+        schema_name = f"acct{schema_name}"
+
+    return schema_name
```
[the remark this answers is not in the document]
Comment (djnakabaale) [reply koku/subs/tasks.py]: Turns out we already have a similar function, [convert_account](https://github.com/project-koku/koku/blob/48641a1d69735dadb102000cdbc9f50534b8cecc/koku/masu/util/common.py#L409). 
I've updated to use this.

File: koku/subs/tasks.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    context = {"schema_name": schema_name}
+    LOG.info(log_json(msg="enable_subs_processing context", context=context))
+
+    # Temporarily disable feature flag check until it is available.
+    # return bool(
+    #     UNLEASH_CLIENT.is_enabled("cost-management.backend.subs-data-processor", context)
+    #     or settings.ENABLE_SUBS_DEBUG
+    # )
+
+    return settings.ENABLE_SUBS_DEBUG
```
[the remark this answers is not in the document]
Comment (djnakabaale) [reply koku/subs/tasks.py]: for now, I've added `return settings.ENABLE_SUBS_EXTRACTION_DEBUG` as fallback

File: deploy/kustomize/patches/worker-subs.yaml [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+          valueFrom:
+            secretKeyRef:
+              key: oci-cli-tenancy
+              name: koku-oci
+        - name: OCI_CLI_KEY_FILE
+          value: ${OCI_CLI_KEY_FILE}
+        - name: OCI_PYTHON_SDK_NO_SERVICE_IMPORTS
+          value: 'true'
+        - name: ENHANCED_ORG_ADMIN
+          value: ${ENHANCED_ORG_ADMIN}
```
[the remark this answers is not in the document]
Comment (djnakabaale) [reply deploy/kustomize/patches/worker-subs.yaml:122]: No we don't! Especially ones related to OCI. 
I've tried to trim it down but please let me know if I've missed any that we might not need.

**Pull request closed without merging**